src/Catalog/Helpers/Db2CatalogProjection.cs

```diff
@@ -19,29 +22,58 @@ public Db2CatalogProjection(PackageContentUriBuilder packageContentUriBuilder)
+            var hideLicenseReport = dataReader.GetBoolean(dataReader.GetOrdinal(Db2CatalogProjectionColumnNames.HideLicenseReport));
 
             var packageContentUri = _packageContentUriBuilder.Build(packageId, normalizedPackageVersion);
+            var deprecationInfo = ReadDeprecationInfoFromDataRecord(dataReader);
 
             return new FeedPackageDetails(
                 packageContentUri,
-                dataRecord.ReadDateTime(Db2CatalogProjectionColumnNames.Created).ForceUtc(),
-                dataRecord.ReadNullableUtcDateTime(Db2CatalogProjectionColumnNames.LastEdited),
-                listed ? dataRecord.ReadDateTime(Db2CatalogProjectionColumnNames.Published).ForceUtc() : Constants.UnpublishedDate,
+                dataReader.ReadDateTime(Db2CatalogProjectionColumnNames.Created).ForceUtc(),
+                dataReader.ReadNullableUtcDateTime(Db2CatalogProjectionColumnNames.LastEdited),
+                listed ? dataReader.ReadDateTime(Db2CatalogProjectionColumnNames.Published).ForceUtc() : Constants.UnpublishedDate,
                 packageId,
                 normalizedPackageVersion,
-                hideLicenseReport ? null : dataRecord[Db2CatalogProjectionColumnNames.LicenseNames]?.ToString(),
-                hideLicenseReport ? null : dataRecord[Db2CatalogProjectionColumnNames.LicenseReportUrl]?.ToString());
+                hideLicenseReport ? null : dataReader[Db2CatalogProjectionColumnNames.LicenseNames]?.ToString(),
+                hideLicenseReport ? null : dataReader[Db2CatalogProjectionColumnNames.LicenseReportUrl]?.ToString(),
+                deprecationInfo);
+        }
+
+        public FeedPackageDeprecationInfo ReadDeprecationInfoFromDataRecord(DbDataReader dataReader)
+        {
+            var deprecationReasons = new List<string>();
+            var deprecationStatus = (PackageDeprecationStatus)dataReader.ReadNullableInt32(Db2CatalogProjectionColumnNames.DeprecationStatus);
+
+            if (deprecationStatus.HasFlag(PackageDeprecationStatus.Legacy))
+            {
+                deprecationReasons.Add(PackageDeprecationStatus.Legacy.ToString());
+            }
+
+            if (deprecationStatus.HasFlag(PackageDeprecationStatus.CriticalBugs))
+            {
+                deprecationReasons.Add(PackageDeprecationStatus.CriticalBugs.ToString());
+            }
+
+            if (deprecationStatus.HasFlag(PackageDeprecationStatus.Other))
+            {
+                deprecationReasons.Add(PackageDeprecationStatus.Other.ToString());
+            }
+
+            return new FeedPackageDeprecationInfo(
+                deprecationReasons,
+                dataReader.ReadNullableString(Db2CatalogProjectionColumnNames.DeprecationMessage),
+                dataReader.ReadNullableString(Db2CatalogProjectionColumnNames.AlternatePackageId),
+                dataReader.ReadNullableString(Db2CatalogProjectionColumnNames.AlternatePackageVersion));
```

Comment:
This is incorrect. We need to pass in the version range, not the version string directly.

This would be either
- `*` when version is null
- `[{version}, {version}]` when a version is specified.

Reply:
Fixed in https://github.com/NuGet/NuGet.Services.Metadata/pull/542/commits/8b9758d296cee6dc79a5f80535fab1f2aad67f7c#diff-3ef253b665c20ee4412e5ad2f110f4f6R89